File: quipuu/crates/scan-source/tests/fixtures/csharp/Pqc.cs
// Fixture: BouncyCastle.Cryptography ML-KEM / ML-DSA calls (release-2.7.0+),
// quipuu scanner tests.
using Org.BouncyCastle.Crypto.Generators;
using Org.BouncyCastle.Crypto.Kems;
using Org.BouncyCastle.Crypto.Parameters;
using Org.BouncyCastle.Crypto.Signers;
using Org.BouncyCastle.Pqc.Crypto.Bike;
using Org.BouncyCastle.Pqc.Crypto.Hqc;
using Org.BouncyCastle.Pqc.Crypto.Lms;
using Org.BouncyCastle.Security;

public class PqcFixture
{
    // CSH-050 / CRYPTO-662 — ML-KEM-768 key-generation parameters
    public static void MlKemKeyPair()
    {
        var random = new SecureRandom();
        var generator = new MLKemKeyPairGenerator();
        generator.Init(new MLKemKeyGenerationParameters(random, MLKemParameters.ml_kem_768));
        var keyPair = generator.GenerateKeyPair();
    }

    // CSH-050 / CRYPTO-664 — parameter set read from a variable, not a literal
    public static void MlKemKeyPairUnattributed(MLKemParameters chosen)
    {
        var random = new SecureRandom();
        var generator = new MLKemKeyPairGenerator();
        generator.Init(new MLKemKeyGenerationParameters(random, chosen));
        var keyPair = generator.GenerateKeyPair();
    }

    // CSH-051 / CRYPTO-666 — ML-DSA-65 key-generation parameters
    public static void MlDsaKeyPair()
    {
        var random = new SecureRandom();
        var generator = new MLDsaKeyPairGenerator();
        generator.Init(new MLDsaKeyGenerationParameters(random, MLDsaParameters.ml_dsa_65));
        var keyPair = generator.GenerateKeyPair();
    }

    // CSH-051 / CRYPTO-667 — HashML-DSA-87 pre-hash variant, same parameter set
    public static void HashMlDsaKeyPair()
    {
        var random = new SecureRandom();
        var generator = new MLDsaKeyPairGenerator();
        generator.Init(new MLDsaKeyGenerationParameters(random, MLDsaParameters.ml_dsa_87_with_sha512));
        var keyPair = generator.GenerateKeyPair();
    }

    // CSH-055 / CRYPTO-827 — ML-KEM-768 encapsulation, operation site
    public static void MlKemEncapsulate(MLKemPublicKeyParameters pub)
    {
        var encapsulator = new MLKemEncapsulator(MLKemParameters.ml_kem_768);
        encapsulator.Init(pub);
    }

    // CSH-056 / CRYPTO-830 — ML-KEM-512 decapsulation, operation site
    public static void MlKemDecapsulate(MLKemPrivateKeyParameters priv)
    {
        var decapsulator = new MLKemDecapsulator(MLKemParameters.ml_kem_512);
        decapsulator.Init(priv);
    }

    // CSH-057 / CRYPTO-836 — ML-DSA-87 sign/verify, operation site
    public static void MlDsaSignVerify(MLDsaPrivateKeyParameters priv)
    {
        var signer = new MLDsaSigner(MLDsaParameters.ml_dsa_87, false);
        signer.Init(true, priv);
    }

    // CSH-080 / CRYPTO-1080 — single-tree LMS key generation (#Y100)
    public static void LmsKeyPair(LmsParameters lmsParameters)
    {
        var random = new SecureRandom();
        var generator = new LmsKeyPairGenerator();
        generator.Init(new LmsKeyGenerationParameters(lmsParameters, random));
        var keyPair = generator.GenerateKeyPair();
    }

    // CSH-081 / CRYPTO-1081 — multi-tree HSS key generation (#Y100)
    public static void HssKeyPair(LmsParameters[] lmsParameters)
    {
        var random = new SecureRandom();
        var generator = new HssKeyPairGenerator();
        generator.Init(new HssKeyGenerationParameters(lmsParameters, random));
        var keyPair = generator.GenerateKeyPair();
    }

    // CSH-083 / CRYPTO-1182 — HQC-128 key-generation parameters (#Y131)
    public static void HqcKeyPair()
    {
        var random = new SecureRandom();
        var keyGenParameters = new HqcKeyGenerationParameters(random, HqcParameters.hqc128);
    }

    // CSH-083 / CRYPTO-1185 — parameter set read from a variable, not a literal
    public static void HqcKeyPairUnattributed(HqcParameters chosen)
    {
        var random = new SecureRandom();
        var keyGenParameters = new HqcKeyGenerationParameters(random, chosen);
    }

    // CSH-084 / CRYPTO-1186 — BIKE-128 key-generation parameters (#Y131)
    public static void BikeKeyPair()
    {
        var random = new SecureRandom();
        var keyGenParameters = new BikeKeyGenerationParameters(random, BikeParameters.bike128);
    }

    // CSH-084 / CRYPTO-1189 — parameter set read from a variable, not a literal
    public static void BikeKeyPairUnattributed(BikeParameters chosen)
    {
        var random = new SecureRandom();
        var keyGenParameters = new BikeKeyGenerationParameters(random, chosen);
    }
}

File: quipuu/crates/scan-source/tests/fixtures/java/Pqc.java
import javax.crypto.Cipher;
import javax.crypto.KEM;
import java.security.KeyPairGenerator;
import java.security.Signature;

/**
 * Fixture: Java PQC API calls (JDK 24+, JEP 496/497), quipuu scanner tests.
 */
public class Pqc {

    // JAV-010 / CRYPTO-216 — ML-KEM-768 keypair generation
    public static void kemKeypair() throws Exception {
        KeyPairGenerator g = KeyPairGenerator.getInstance("ML-KEM-768");
    }

    // JAV-010 / CRYPTO-219 — ML-DSA-65 keypair generation
    public static void dsaKeypair() throws Exception {
        KeyPairGenerator g = KeyPairGenerator.getInstance("ML-DSA-65");
    }

    // JAV-090 / CRYPTO-225 — ML-DSA-65 signature
    public static void dsaSignature() throws Exception {
        Signature s = Signature.getInstance("ML-DSA-65");
    }

    // JAV-040 / CRYPTO-228 — ML-KEM-768 encapsulation object
    public static void kem() throws Exception {
        KEM kem = KEM.getInstance("ML-KEM-768");
    }

    // JAV-041 / CRYPTO-1168 — ML-KEM encapsulator operation (parameter set
    // not traceable to this call site; see java.toml's comment above JAV-041)
    public static void kemEncapsulate(java.security.PublicKey pub) throws Exception {
        KEM kem = KEM.getInstance("ML-KEM-768");
        KEM.Encapsulator enc = kem.newEncapsulator(pub);
    }

    // JAV-042 / CRYPTO-1169 — ML-KEM decapsulator operation (parameter set
    // not traceable to this call site; see java.toml's comment above JAV-042)
    public static void kemDecapsulate(java.security.PrivateKey priv) throws Exception {
        KEM kem = KEM.getInstance("ML-KEM-768");
        KEM.Decapsulator dec = kem.newDecapsulator(priv);
    }

    // JAV-010 / CRYPTO-770 — SLH-DSA-SHA2-128S keypair generation (BouncyCastle)
    public static void slhDsaKeypair() throws Exception {
        KeyPairGenerator g = KeyPairGenerator.getInstance("SLH-DSA-SHA2-128S", "BC");
    }

    // JAV-010 / CRYPTO-782 — SLH-DSA family-generic name, no parameter set stated
    public static void slhDsaKeypairUnattributed() throws Exception {
        KeyPairGenerator g = KeyPairGenerator.getInstance("SLH-DSA", "BC");
    }

    // JAV-090 / CRYPTO-790 — SLH-DSA-SHAKE-128S signature
    public static void slhDsaSignature() throws Exception {
        Signature s = Signature.getInstance("SLH-DSA-SHAKE-128S", "BC");
    }

    // JAV-010 / CRYPTO-1006 — HQC-128 keypair generation (BouncyCastle)
    public static void hqcKeypair() throws Exception {
        KeyPairGenerator g = KeyPairGenerator.getInstance("HQC-128", "BCPQC");
    }

    // JAV-010 / CRYPTO-1009 — HQC family-generic name, no parameter set stated
    public static void hqcKeypairUnattributed() throws Exception {
        KeyPairGenerator g = KeyPairGenerator.getInstance("HQC", "BCPQC");
    }

    // JAV-040 / CRYPTO-1011 — HQC-192 encapsulation object
    public static void hqcKem() throws Exception {
        KEM kem = KEM.getInstance("HQC-192");
    }

    // JAV-001 / CRYPTO-1004 — HQC-256 cipher
    public static void hqcCipher() throws Exception {
        Cipher c = Cipher.getInstance("HQC-256");
    }

    // JAV-001 / CRYPTO-1014 — BIKE128 cipher (BouncyCastle)
    public static void bikeCipher() throws Exception {
        Cipher c = Cipher.getInstance("BIKE128");
    }

    // JAV-010 / CRYPTO-1018 — BIKE family-generic name, no parameter set stated
    public static void bikeKeypairUnattributed() throws Exception {
        KeyPairGenerator g = KeyPairGenerator.getInstance("BIKE", "BCPQC");
    }

    // JAV-010 / CRYPTO-1019 — Classic McEliece keypair generation, family sentinel
    public static void cmceKeypair() throws Exception {
        KeyPairGenerator g = KeyPairGenerator.getInstance("mceliece6960119", "BCPQC");
    }

    // JAV-040 / CRYPTO-1020 — Classic McEliece encapsulation object, family sentinel
    public static void cmceKem() throws Exception {
        KEM kem = KEM.getInstance("CMCE");
    }

    // JAV-010 / CRYPTO-1021 — XMSS (single-tree) keypair generation
    public static void xmssKeypair() throws Exception {
        KeyPairGenerator g = KeyPairGenerator.getInstance("XMSS", "BCPQC");
    }

    // JAV-010 / CRYPTO-1022 — XMSSMT (multi-tree) keypair generation
    public static void xmssMtKeypair() throws Exception {
        KeyPairGenerator g = KeyPairGenerator.getInstance("XMSSMT", "BCPQC");
    }

    // JAV-090 / CRYPTO-1025 — XMSS signature, digest-qualified name
    public static void xmssSignature() throws Exception {
        Signature s = Signature.getInstance("XMSS-SHA256", "BCPQC");
    }

    // JAV-090 / CRYPTO-1029 — XMSSMT signature, digest-qualified name
    public static void xmssMtSignature() throws Exception {
        Signature s = Signature.getInstance("XMSSMT-SHA256", "BCPQC");
    }
}
